File: sdks/go/admin/orgs.go
package admin

import (
	"fmt"
	"net/url"
	"strconv"
)

// Org represents a Nucleus organization returned by the Admin API.
type Org struct {
	ID        string                 `json:"id"`
	Name      string                 `json:"name"`
	Slug      string                 `json:"slug"`
	LogoURL   string                 `json:"logo_url,omitempty"`
	Metadata  map[string]interface{} `json:"metadata,omitempty"`
	CreatedAt string                 `json:"created_at"`
	UpdatedAt string                 `json:"updated_at"`
}

// ListOrgsParams controls pagination for listing organizations.
type ListOrgsParams struct {
	Limit  int
	Cursor string
}

// OrgsAPI provides access to the Nucleus Admin Orgs endpoints.
type OrgsAPI struct {
	client *HTTPClient
}

// NewOrgsAPI creates a new OrgsAPI instance.
func NewOrgsAPI(client *HTTPClient) *OrgsAPI {
	return &OrgsAPI{client: client}
}

// Get retrieves a single organization by ID.
func (a *OrgsAPI) Get(orgID string) (*Org, error) {
	var org Org
	if err := a.client.Get(fmt.Sprintf("/orgs/%s", orgID), &org); err != nil {
		return nil, err
	}
	return &org, nil
}

// List retrieves a paginated list of organizations.
func (a *OrgsAPI) List(params *ListOrgsParams) (*PaginatedResponse[Org], error) {
	query := url.Values{}
	if params != nil {
		if params.Limit > 0 {
			query.Set("limit", strconv.Itoa(params.Limit))
		}
		if params.Cursor != "" {
			query.Set("cursor", params.Cursor)
		}
	}

	path := "/orgs"
	if len(query) > 0 {
		path += "?" + query.Encode()
	}

	var resp PaginatedResponse[Org]
	if err := a.client.Get(path, &resp); err != nil {
		return nil, err
	}
	return &resp, nil
}

File: sdks/rust/src/admin/orgs.rs
use super::{HttpClient, PaginatedResponse};
use crate::verify::NucleusError;
use serde::Deserialize;
use std::sync::Arc;

/// A Nucleus organisation returned by the Admin API.
#[derive(Debug, Clone, Deserialize)]
pub struct NucleusOrg {
    pub id: String,
    pub name: String,
    pub slug: String,
    pub logo_url: Option<String>,
    pub metadata: serde_json::Value,
    pub created_at: String,
    pub updated_at: String,
}

/// Optional parameters for listing organisations.
#[derive(Debug, Default)]
pub struct ListOrgsParams {
    pub limit: Option<u32>,
    pub cursor: Option<String>,
}

/// Admin Orgs API.
#[derive(Clone)]
pub struct OrgsApi {
    http: Arc<HttpClient>,
}

impl OrgsApi {
    pub(crate) fn new(http: Arc<HttpClient>) -> Self {
        Self { http }
    }

    /// Get a single organisation by ID.
    pub async fn get(&self, org_id: &str) -> Result<NucleusOrg, NucleusError> {
        self.http.get(&format!("/orgs/{org_id}")).await
    }

    /// List organisations with optional pagination.
    pub async fn list(
        &self,
        params: Option<ListOrgsParams>,
    ) -> Result<PaginatedResponse<NucleusOrg>, NucleusError> {
        let mut query_parts: Vec<String> = Vec::new();
        if let Some(p) = &params {
            if let Some(limit) = p.limit {
                query_parts.push(format!("limit={limit}"));
            }
            if let Some(cursor) = &p.cursor {
                query_parts.push(format!("cursor={cursor}"));
            }
        }

        let path = if query_parts.is_empty() {
            "/orgs".to_string()
        } else {
            format!("/orgs?{}", query_parts.join("&"))
        };

        self.http.get(&path).await
    }
}
